airbyte-webapp/src/hooks/services/ConfirmationModal/ConfirmationModalService.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useContext, useEffect, useMemo } from "react";
+
+import { ConfirmationModal } from "components/ConfirmationModal";
+
+import useTypesafeReducer from "hooks/useTypesafeReducer";
+
+import { ConfirmationModalOptions, ConfirmationModalServiceApi, ConfirmationModalState } from "./types";
+import { actions, initialState, confirmationModalServiceReducer } from "./reducer";
+
+const ConfirmationModalServiceContext = React.createContext<ConfirmationModalServiceApi | undefined>(undefined);
+
+export const useConfirmationModalService: (
+  confirmationModal?: ConfirmationModalOptions,
+  dependencies?: []
+) => {
+  openConfirmationModal: (confirmationModal: ConfirmationModalOptions) => void;
+  closeConfirmationModal: () => void;
+} = (confirmationModal, dependencies) => {
+  const confirmationModalService = useContext(ConfirmationModalServiceContext);
+  if (!confirmationModalService) {
+    throw new Error("useConfirmationModalService must be used within a ConfirmationModalService.");
+  }
+
+  useEffect(() => {
+    if (confirmationModal) {
+      confirmationModalService.openConfirmationModal(confirmationModal);
+    }
+    return () => {
+      if (confirmationModal) {
+        confirmationModalService.closeConfirmationModal();
+      }
+    };
+    // eslint-disable-next-line
+  }, [confirmationModal, confirmationModalService, ...(dependencies ?? [])]);
+
+  return {
+    openConfirmationModal: confirmationModalService.openConfirmationModal,
+    closeConfirmationModal: confirmationModalService.closeConfirmationModal,
+  };
+};
+
+export const ConfirmationModalService = React.memo(({ children }: { children: React.ReactNode }) => {
```

Comment:
Given the use of services elsewhere in the app, perhaps this should just be called `ConfirmationModal`?

Reply:
Nitpick: I'm not sure how much, if anything, React.memo gets us here.
I haven't used it in the past but it seems to allow for some optimization around props-changing but since it uses shallow compares I don't know if it can really do much with `children` here in terms of optimization.